tests/test_recreation.py

```diff
@@ -1187,6 +1187,51 @@ def test_bad_predictor_table_header(self):
             actual_messages.add(error_strings)
         self.assertTrue(expected_message in actual_messages)
 
+    def test_validate_predictor_types(self):
+        """Recreation Validate: assert error on invalid type value"""
+        from natcap.invest.recreation import recmodel_client
+
+        predictor_id = 'dem90m'
+        # include trailing whitespace in the type, this should pass
+        raster_path = os.path.join(SAMPLE_DATA, 'predictors/dem90m_coarse.tif')
+        table_path = os.path.join(self.workspace_dir, 'table.csv')
+        with open(table_path, 'w') as file:
+            file.write('id,path,type\n')
+            file.write(f'{predictor_id},{raster_path},raster_mean \n')
+
+        # include a typo in the type, this should fail
+        bad_table_path = os.path.join(self.workspace_dir, 'bad_table.csv')
+        with open(bad_table_path, 'w') as file:
+            file.write('id,path,type\n')
+            file.write(f'{predictor_id},{raster_path},raster?mean\n')
+
+        args = {
+            'aoi_path': os.path.join(SAMPLE_DATA, 'andros_aoi.shp'),
+            'cell_size': 40000.0,
+            'compute_regression': True,
+            'start_year': '2005',
+            'end_year': '2014',
+            'grid_aoi': False,
+            'predictor_table_path': table_path,
+            'workspace_dir': self.workspace_dir,
+        }
+
+        # there should be no error when the type has trailing whitespace
+        recmodel_client.execute(args)
+        output_path = os.path.join(self.workspace_dir, 'regression_coefficients.txt')
+        print('regression_coefficients.txt:')

```

Comment:
Should this be a logging statement? Is it needed here?

Reply:
oops! Left that in by mistake